--- drivers/platform/x86/fujitsu-tablet.c.orig	2014-01-20 11:40:07.000000000 +0900
+++ drivers/platform/x86/fujitsu-tablet.c	2014-04-19 21:24:49.039693219 +0900
@@ -90,6 +90,25 @@
 	KEY_LEFTALT
 };
 
+static unsigned short keymap_Lifebook_LOOXU[KEYMAP_LEN] __initdata = {
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_PROG1,
+	KEY_PROG2,
+	KEY_DIRECTION,
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_SCROLLUP,
+	KEY_SCROLLDOWN,
+	KEY_RESERVED,
+	KEY_RESERVED,
+	KEY_F13,
+	KEY_F14
+};
+
 static unsigned short keymap_Stylistic_Tseries[KEYMAP_LEN] __initdata = {
 	KEY_RESERVED,
 	KEY_RESERVED,
@@ -337,6 +356,15 @@
 		.driver_data = keymap_Lifebook_U810
 	},
 	{
+		.callback = fujitsu_dmi_lifebook,
+		.ident = "Fujitsu LOOX U/C40",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "FUJITSU"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "FMVLUC40N")
+		},
+		.driver_data = keymap_Lifebook_LOOXU
+	},
+	{
 		.callback = fujitsu_dmi_stylistic,
 		.ident = "Fujitsu Siemens Stylistic ST5xxx Series",
 		.matches = {
