--- drivers/platform/x86/fujitsu-tablet.c.orig	2014-01-20 11:40:07.000000000 +0900
+++ drivers/platform/x86/fujitsu-tablet.c	2014-04-12 10:23:32.820304810 +0900
@@ -337,6 +337,15 @@
 		.driver_data = keymap_Lifebook_U810
 	},
 	{
+		.callback = fujitsu_dmi_lifebook,
+		.ident = "Fujitsu LOOX U/C40",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "FUJITSU"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "FMVLUC40N")
+		},
+		.driver_data = keymap_Lifebook_U810
+	},
+	{
 		.callback = fujitsu_dmi_stylistic,
 		.ident = "Fujitsu Siemens Stylistic ST5xxx Series",
 		.matches = {
