diff -ru a/drivers/gpu/drm/i915/i915_params.c b/drivers/gpu/drm/i915/i915_params.c
--- a/drivers/gpu/drm/i915/i915_params.c	2018-01-29 06:20:33.000000000 +0900
+++ b/drivers/gpu/drm/i915/i915_params.c	2018-05-03 15:22:06.680610985 +0900
@@ -190,3 +190,12 @@
 
 i915_param_named(enable_gvt, bool, 0400,
 	"Enable support for Intel GVT-g graphics virtualization host support(default:false)");
+
+i915_param_named(min_clock_force, int, 0400,
+	"Set GPU minimum clock force (-1=default)");
+i915_param_named(max_clock_force, int, 0400,
+	"Set GPU maximum clock force (-1=default)");
+i915_param_named(eff_clock_force, int, 0400,
+	"Set GPU efficient clock force (-1=default)");
+i915_param_named(guard_clock_force, int, 0400,
+	"Set GPU guard clock force (-1=default)");
diff -ru a/drivers/gpu/drm/i915/i915_params.h b/drivers/gpu/drm/i915/i915_params.h
--- a/drivers/gpu/drm/i915/i915_params.h	2018-01-29 06:20:33.000000000 +0900
+++ b/drivers/gpu/drm/i915/i915_params.h	2018-05-03 15:23:47.868594557 +0900
@@ -67,7 +67,11 @@
 	param(bool, nuclear_pageflip, false) \
 	param(bool, enable_dp_mst, true) \
 	param(bool, enable_dpcd_backlight, false) \
-	param(bool, enable_gvt, false)
+	param(bool, enable_gvt, false)	\
+	param(int, min_clock_force, -1)	\
+	param(int, max_clock_force, -1)	\
+	param(int, eff_clock_force, -1)	\
+	param(int, guard_clock_force, -1)
 
 #define MEMBER(T, member, ...) T member;
 struct i915_params {
diff -ru a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
--- a/drivers/gpu/drm/i915/intel_pm.c	2018-01-29 06:20:33.000000000 +0900
+++ b/drivers/gpu/drm/i915/intel_pm.c	2018-05-03 15:46:59.976260347 +0900
@@ -6521,15 +6521,39 @@
 	/* static values from HW: RP0 > RP1 > RPn (min_freq) */
 	if (IS_GEN9_LP(dev_priv)) {
 		u32 rp_state_cap = I915_READ(BXT_RP_STATE_CAP);
-		rps->rp0_freq = (rp_state_cap >> 16) & 0xff;
-		rps->rp1_freq = (rp_state_cap >>  8) & 0xff;
-		rps->min_freq = (rp_state_cap >>  0) & 0xff;
+		if (i915_modparams.max_clock_force > 0)
+			rps->rp0_freq = i915_modparams.max_clock_force / 50;
+		else
+			rps->rp0_freq = (rp_state_cap >> 16) & 0xff;
+		if (i915_modparams.eff_clock_force > 0)
+			rps->rp1_freq = i915_modparams.eff_clock_force / 50;
+		else
+			rps->rp1_freq = (rp_state_cap >>  8) & 0xff;
+		if (i915_modparams.min_clock_force > 0)
+			rps->min_freq = i915_modparams.min_clock_force / 50;
+		else
+			rps->min_freq = (rp_state_cap >>  0) & 0xff;
 	} else {
 		u32 rp_state_cap = I915_READ(GEN6_RP_STATE_CAP);
-		rps->rp0_freq = (rp_state_cap >>  0) & 0xff;
-		rps->rp1_freq = (rp_state_cap >>  8) & 0xff;
-		rps->min_freq = (rp_state_cap >> 16) & 0xff;
-	}
+		if (i915_modparams.max_clock_force > 0)
+			rps->rp0_freq = i915_modparams.max_clock_force / 50;
+		else
+			rps->rp0_freq = (rp_state_cap >>  0) & 0xff;
+		if (i915_modparams.eff_clock_force > 0)
+			rps->rp1_freq = i915_modparams.eff_clock_force / 50;
+		else
+			rps->rp1_freq = (rp_state_cap >>  8) & 0xff;
+		if (i915_modparams.min_clock_force > 0)
+			rps->min_freq = i915_modparams.min_clock_force / 50;
+		else
+			rps->min_freq = (rp_state_cap >> 16) & 0xff;
+	}
+	if (rps->min_freq > rps->rp0_freq)
+		rps->min_freq = rps->rp0_freq;
+	if (rps->rp1_freq > rps->rp0_freq)
+		rps->rp1_freq = rps->rp0_freq;
+	if (rps->rp1_freq < rps->min_freq)
+		rps->rp1_freq = rps->min_freq;
 	/* hw_max = RP0 until we check for overclocking */
 	rps->max_freq = rps->rp0_freq;
 
@@ -7158,22 +7182,35 @@
 	}
 	DRM_DEBUG_DRIVER("DDR speed: %d MHz\n", dev_priv->mem_freq);
 
-	rps->max_freq = valleyview_rps_max_freq(dev_priv);
+	if (i915_modparams.max_clock_force > 0)
+		rps->max_freq = intel_freq_opcode(dev_priv, i915_modparams.max_clock_force);
+	else
+		rps->max_freq = valleyview_rps_max_freq(dev_priv);
 	rps->rp0_freq = rps->max_freq;
 	DRM_DEBUG_DRIVER("max GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->max_freq),
 			 rps->max_freq);
 
-	rps->efficient_freq = valleyview_rps_rpe_freq(dev_priv);
+	if (i915_modparams.eff_clock_force > 0)
+		rps->efficient_freq = intel_freq_opcode(dev_priv, i915_modparams.eff_clock_force);
+	else
+		rps->efficient_freq = valleyview_rps_rpe_freq(dev_priv);
 	DRM_DEBUG_DRIVER("RPe GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->efficient_freq),
 			 rps->efficient_freq);
 
-	rps->rp1_freq = valleyview_rps_guar_freq(dev_priv);
+	if (i915_modparams.guard_clock_force > 0)
+		rps->rp1_freq =	intel_freq_opcode(dev_priv, i915_modparams.guard_clock_force);
+	else
+		rps->rp1_freq = valleyview_rps_guar_freq(dev_priv);
 	DRM_DEBUG_DRIVER("RP1(Guar Freq) GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->rp1_freq),
 			 rps->rp1_freq);
 
+	if (i915_modparams.min_clock_force > 0)
+		rps->min_freq = intel_freq_opcode(dev_priv, i915_modparams.min_clock_force);
+	else
+		rps->min_freq = valleyview_rps_min_freq(dev_priv);
 	rps->min_freq = valleyview_rps_min_freq(dev_priv);
 	DRM_DEBUG_DRIVER("min GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->min_freq),
@@ -7203,23 +7240,35 @@
 	}
 	DRM_DEBUG_DRIVER("DDR speed: %d MHz\n", dev_priv->mem_freq);
 
-	rps->max_freq = cherryview_rps_max_freq(dev_priv);
+	if (i915_modparams.max_clock_force > 0)
+		rps->max_freq = intel_freq_opcode(dev_priv, i915_modparams.max_clock_force);
+	else
+		rps->max_freq = cherryview_rps_max_freq(dev_priv);
 	rps->rp0_freq = rps->max_freq;
 	DRM_DEBUG_DRIVER("max GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->max_freq),
 			 rps->max_freq);
 
-	rps->efficient_freq = cherryview_rps_rpe_freq(dev_priv);
+	if (i915_modparams.eff_clock_force > 0)
+		rps->efficient_freq = intel_freq_opcode(dev_priv, i915_modparams.eff_clock_force);
+	else
+		rps->efficient_freq = cherryview_rps_rpe_freq(dev_priv);
 	DRM_DEBUG_DRIVER("RPe GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->efficient_freq),
 			 rps->efficient_freq);
 
-	rps->rp1_freq = cherryview_rps_guar_freq(dev_priv);
+	if (i915_modparams.guard_clock_force > 0)
+		rps->rp1_freq = intel_freq_opcode(dev_priv, i915_modparams.guard_clock_force);
+	else
+		rps->rp1_freq = cherryview_rps_guar_freq(dev_priv);
 	DRM_DEBUG_DRIVER("RP1(Guar) GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->rp1_freq),
 			 rps->rp1_freq);
 
-	rps->min_freq = cherryview_rps_min_freq(dev_priv);
+	if (i915_modparams.min_clock_force > 0)
+		rps->min_freq =	intel_freq_opcode(dev_priv, i915_modparams.min_clock_force);
+	else
+		rps->min_freq = cherryview_rps_min_freq(dev_priv);
 	DRM_DEBUG_DRIVER("min GPU freq: %d MHz (%u)\n",
 			 intel_gpu_freq(dev_priv, rps->min_freq),
 			 rps->min_freq);
