--- a/sound/soc/codecs/rt5670.c	2018-01-29 06:20:33.000000000 +0900
+++ b/sound/soc/codecs/rt5670.c	2018-05-04 03:30:17.540020550 +0900
@@ -2924,6 +2924,17 @@
 						 RT5670_DEV_GPIO |
 						 RT5670_JD_MODE3),
 	},
+	{
+		.callback = rt5670_quirk_cb,
+		.ident = "Kingjim XMC10",
+		.matches = {
+			DMI_MATCH(DMI_BOARD_NAME, "Cherry Trail FFD"),
+		},
+		.driver_data = (unsigned long *)(RT5670_DMIC_EN |
+						 RT5670_DMIC1_IN2P |
+						 RT5670_DEV_GPIO |
+						 RT5670_JD_MODE1),
+	},
 	{}
 };
 
