--- a/drivers/mmc/core/core.c	2017-09-04 05:56:17.000000000 +0900
+++ b/drivers/mmc/core/core.c	2018-01-13 00:38:11.425255689 +0900
@@ -2749,6 +2749,10 @@
 	case PM_HIBERNATION_PREPARE:
 	case PM_SUSPEND_PREPARE:
 	case PM_RESTORE_PREPARE:
+		if (host->slot.cd_irq >= 0) {
+			free_irq(host->slot.cd_irq, host);
+			host->slot.cd_irq = -EINVAL;
+		}
 		spin_lock_irqsave(&host->lock, flags);
 		host->rescan_disable = 1;
 		spin_unlock_irqrestore(&host->lock, flags);
@@ -2775,6 +2779,9 @@
 	case PM_POST_SUSPEND:
 	case PM_POST_HIBERNATION:
 	case PM_POST_RESTORE:
+		if (host->slot.cd_irq < 0) {
+			mmc_gpiod_request_cd_irq(host);
+		}
 
 		spin_lock_irqsave(&host->lock, flags);
 		host->rescan_disable = 0;
