diff -ru a/drivers/staging/rtl8723bs/core/rtw_recv.c b/drivers/staging/rtl8723bs/core/rtw_recv.c
--- a/drivers/staging/rtl8723bs/core/rtw_recv.c	2018-08-13 05:41:04.000000000 +0900
+++ b/drivers/staging/rtl8723bs/core/rtw_recv.c	2019-02-05 21:39:22.123531832 +0900
@@ -2496,6 +2496,7 @@
 		DBG_COUNTER(padapter->rx_logs.core_rx_post_indicate_err);
 		goto _recv_data_drop;
 	}
+	return ret;
 
 _recv_data_drop:
 	precvpriv->rx_drop++;
diff -ru a/drivers/staging/rtl8723bs/hal/rtl8723b_cmd.c b/drivers/staging/rtl8723bs/hal/rtl8723b_cmd.c
--- a/drivers/staging/rtl8723bs/hal/rtl8723b_cmd.c	2018-08-13 05:41:04.000000000 +0900
+++ b/drivers/staging/rtl8723bs/hal/rtl8723b_cmd.c	2019-02-05 21:39:22.127531952 +0900
@@ -1993,7 +1993,7 @@
 #endif
 			DLBcnCount++;
 			do {
-				yield();
+				cond_resched();
 				/* mdelay(10); */
 				/*  check rsvd page download OK. */
 				rtw_hal_get_hwreg(padapter, HW_VAR_BCN_VALID, (u8 *)(&bcn_valid));
@@ -2305,7 +2305,7 @@
 		SetFwRsvdPagePkt_BTCoex(padapter);
 		DLBcnCount++;
 		do {
-			yield();
+			cond_resched();
 /* 			mdelay(10); */
 			/*  check rsvd page download OK. */
 			rtw_hal_get_hwreg(padapter, HW_VAR_BCN_VALID, &bcn_valid);
diff -ru a/drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c b/drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c
--- a/drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c	2018-08-13 05:41:04.000000000 +0900
+++ b/drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c	2019-02-05 21:39:22.127531952 +0900
@@ -234,7 +234,7 @@
 		value32 = rtw_read32(adapter, REG_MCUFWDL);
 		if (value32 & FWDL_ChkSum_rpt || adapter->bSurpriseRemoved || adapter->bDriverStopped)
 			break;
-		yield();
+		cond_resched();
 	} while (jiffies_to_msecs(jiffies-start) < timeout_ms || cnt < min_cnt);
 
 	if (!(value32 & FWDL_ChkSum_rpt)) {
@@ -284,7 +284,7 @@
 		value32 = rtw_read32(adapter, REG_MCUFWDL);
 		if (value32 & WINTINI_RDY || adapter->bSurpriseRemoved || adapter->bDriverStopped)
 			break;
-		yield();
+		cond_resched();
 	} while (jiffies_to_msecs(jiffies - start) < timeout_ms || cnt < min_cnt);
 
 	if (!(value32 & WINTINI_RDY)) {
diff -ru a/drivers/staging/rtl8723bs/hal/rtl8723bs_xmit.c b/drivers/staging/rtl8723bs/hal/rtl8723bs_xmit.c
--- a/drivers/staging/rtl8723bs/hal/rtl8723bs_xmit.c	2018-08-13 05:41:04.000000000 +0900
+++ b/drivers/staging/rtl8723bs/hal/rtl8723bs_xmit.c	2019-02-05 21:39:22.131532072 +0900
@@ -391,7 +391,7 @@
 				rtw_free_xmitframe(pxmitpriv, pframe);
 				pxmitbuf->priv_data = NULL;
 				enqueue_pending_xmitbuf(pxmitpriv, pxmitbuf);
-				yield();
+				cond_resched();
 			} else
 				rtw_free_xmitbuf(pxmitpriv, pxmitbuf);
 			pxmitbuf = NULL;
@@ -459,7 +459,7 @@
 		if (padapter->registrypriv.wifi_spec)
 			msleep(1);
 		else
-			yield();
+			cond_resched();
 		goto next;
 	}
 
