diff -ru a/drivers/staging/rtl8723bs/core/rtw_recv.c b/drivers/staging/rtl8723bs/core/rtw_recv.c
--- a/drivers/staging/rtl8723bs/core/rtw_recv.c	2018-08-13 05:41:04.000000000 +0900
+++ b/drivers/staging/rtl8723bs/core/rtw_recv.c	2019-02-05 21:39:22.123531832 +0900
@@ -2496,6 +2496,7 @@
 		DBG_COUNTER(padapter->rx_logs.core_rx_post_indicate_err);
 		goto _recv_data_drop;
 	}
+	return ret;
 
 _recv_data_drop:
 	precvpriv->rx_drop++;
